contracts/parameterized/Parameterized.sol

```diff
@@ -0,0 +1,134 @@
+  string Name = "Parameterized";
+  /* solhint-enable */
+
+  function getRegistryInformation() external override view returns (RegisteredTransfer memory) {
+    RegisteredTransfer memory info = RegisteredTransfer({
+      name: Name,
+      stateEncoding: StateEncoding,
+      resolverEncoding: ResolverEncoding,
+      definition: address(this)
+    });
+    return info;
+  }
+
+  /// @notice Creates a parameterized payment from the payer (creator) to the payee (resolver)
+  /// @param encodedBalance balance the creator/payer is putting into the transfer
+  /// @param encodedState encoded transfer parameters
+  /// @dev encoded parameter data contains the receiver address, the expiration, the UUID, and the rate
+  function create(
+    bytes calldata encodedBalance,
+    bytes calldata encodedState
+  ) external override view returns (bool) {
+    // Unpack balance & state
+    Balance memory balance = abi.decode(encodedBalance, (Balance));
+    TransferState memory state = abi.decode(encodedState, (TransferState));
+    Rate memory rate = state.rate;
+
+    // Receiver should not have an initial balance within the transfer
+    require(balance.amount[1] == 0, "Cannot create parameterized payment with nonzero recipient init balance");
+
+    // Parameterized payment checks
+    require(rate.deltaAmount >= 1, "Per-unit amount must be at least 1 wei");
+    require(rate.deltaTime >= 1, "Per-unit time must be at least 1 second");
+    require(state.expiration > block.timestamp + 3 days, "Expiration must be at least 3 days in the future.");
+
+    // Sanity checks
+    require(state.receiver != address(0x0), "Receiver address cannot be the zero address!");
+    require(state.UUID != bytes32(0), "UUID cannot be null.");
+
+    return true;
+  }
+
+  /// @notice Resolves a parameterized payment from the payer (creator) to the payee (resolver)
+  /// @param encodedBalance balance the creator/payer is putting into the transfer; same as in create
+  /// @param encodedState encoded transfer parameters
+  /// @param encodedResolver contains the resolution data (UUID, payment amount) and a signature of said data
+  function resolve(
+    bytes calldata encodedBalance,
+    bytes calldata encodedState,
+    bytes calldata encodedResolver
+  ) external override view returns (Balance memory) {
+    // Unpack balance, state, resolver, & resolverdata
+    Balance memory balance = abi.decode(encodedBalance, (Balance));
+    TransferState memory state = abi.decode(encodedState, (TransferState));
+    TransferResolver memory resolver = abi.decode(encodedResolver, (TransferResolver));
+    ResolverData memory data = resolver.data;
+
+    // Signature check
+    bytes32 hashedData = keccak256(abi.encode(data));
+
+    // solhint-disable-next-line
+    require(state.receiver == LibChannelCrypto.recoverChannelMessageSigner(hashedData, resolver.payeeSignature), "Receiver signature did not match.");
```

Comment:
May also need an ability to add a cancelling action so the payment can be removed from the channel. This could happen post-expiry only if needed, but is recommended

Reply:
Changed this to require(data.paymentAmountTaken == 0 || block.timestamp < state.expiration, "Payment expired!");